### PR TITLE
Issue #164

### PR DESCRIPTION
**Pull request by gregw**
Work around for issue #164 by editing the generated URLs.   Tested against guestbook stage 3

### CODE REVIEW

**Comment by gregw** [comment]
Ooops forgot to add the environment variable to disable!   Tomorrow...

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java [hunk 1]
```diff
@@ -159,6 +166,36 @@ public VmApiProxyDelegate() {
           "complete. Service bridge status code: %s; response " +
           "content-length: %s. Took %s ms.", packageName, methodName, requestData.length, 200,
           responseData.length, (end - start)));
+      
+      // HACK TO FIX USER_SERVICE BUG
+      if ("user".equals(packageName)) {
+        String host = (String) environment.getAttributes().get("com.google.appengine.runtime.host");
+        String https = (String) environment.getAttributes().get("com.google.appengine.runtime.https");
+        if (host != null && host.length() > 0) {
```
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: How or where these 2 attributes are set?

Comment (gregw) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: @ludoch I added those attributes to the AttributeMapping enum in VmApiProxyEnvironment

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java [hunk 2]
```diff
@@ -207,7 +205,16 @@
     APPSERVER_TASK_BNS(
         "X-AppEngine-Appserver-Task-Bns",
         "com.google.appengine.runtime.appserver_task_bns",
-        "", true);
+        "", true),
+    HTTPS(
```
Comment (gregw) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java]: @ludoch the extra attributes are set here

File: jetty9/pom.xml [hunk 3]
```diff
@@ -27,12 +27,6 @@
       <version>${project.version}</version>
       <type>tar.gz</type>
     </dependency>
-    <dependency>
```
Comment (gregw) [inline jetty9/pom.xml]: This is an unrelated fix.  Should be in async-support branch anyway to avoid unneccessary transitive dependencies.

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java (hunk 2 above)
Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyEnvironment.java]: Right.... But how is setting them? I am not sure they are set in the incoming request sent from the app server bridge (outside the jvm?)

File: jetty9/pom.xml (hunk 3 above)
Comment (gregw) [inline jetty9/pom.xml]: @ludoch The Host header is required for HTTP/1.1 and commonly sent with HTTP/1.0.   The X-AppEngine-Https header is used elsewhere in the code to determine if the request is secure.  You can see both headers in a [dump](https://jetty9-work.appspot.com/dump/info)

Comment (gregw) [inline jetty9/pom.xml]: I just tried to hit the site without a Host header and it failed because it is a virtual host sharing an IP address.  So no host header will always result in a failed request.

Comment (ludoch) [inline jetty9/pom.xml]: Perfect... remaining issue is the flag to turn this off when we can

File: appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java [hunk 4]
```diff
@@ -159,6 +166,37 @@ public VmApiProxyDelegate() {
           "complete. Service bridge status code: %s; response " +
           "content-length: %s. Took %s ms.", packageName, methodName, requestData.length, 200,
           responseData.length, (end - start)));
+      
+      // HACK TO FIX USER_SERVICE ISSUE #164
+      if ("user".equals(packageName)) {
+        boolean userservicefix = !Boolean.getBoolean("UserService164");
```
Comment (gregw) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: System prop to disable has been added.  @ludoch maybe a bad name?

Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: yes.... Maybe useLocalUserService ?
And true by default

Comment (gregw) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: Well it is not exactly "Local", as we still call the RPC, we just correct the results.   Perhaps UserServiceLocalSchemeHost?

Comment (ludoch) [inline appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java]: ok

**Comment by ludoch** [comment]
LGTM
